src/ui/dialogs/contactsearch.ts

```diff
@@ -0,0 +1,278 @@
+    {
+
+        header.push(Translation.t('First')); //first
+        header.push(Translation.t('Last')); //last
+        header.push(Translation.t('Nick')); //nick
+        header.push(Translation.t('Email')); //email
+
+        simple=true;
+    }
+    else
+        if ($(result).find('query').children('x').length>0)//Extended results
+        {
+            items=$(result).find('query').children('x').children('item');
+            if (items.length===0)
+            {
+                //NO RESULTS FOUND!
+                dialog.getDom().find('#jsxc-search-content').text(Translation.t('no_results'));
+                return;
+            }
+            fields = $(result).find('query').find('reported').children('field');
+            for (let field of fields)
+            {
+                header.push($(field).attr('label'));
+            }
+        }
+        else
+        {
+            //NO RESULTS FOUND!
+            dialog.getDom().find('#jsxc-search-content').text(Translation.t('no_results'));
+            return;
+        }
+
+    let html = null;
+    if (simple)
+    {
+        html=getTabeleForSimpleResult(header,items);
+    }
+    else
+    {
+        html=getTableForExtendedResults(header,items,fields);
+    }
+
+    dialog.getDom().find('#jsxc-search-content').empty().append(html);
+    dialog.getDom().find('#jsxc-search-content').append(submitButton);
+
+    $('#jsxc-searchresults-button-add').on('click',function(e)
+    {
+        let chboxarr = $('input[name="jsxc-add-contact"]:checked');
+        if (chboxarr.length>0)
+        {
+            for (let chboxitm of chboxarr)
+            {
+                let val = Number($(chboxitm).val());
+                let jid = simple?new JID($(items[val]).children('nick').text()+'@'+account.getJID().domain):new JID($(items[val]).find('field[var="jid"]').text());
+                let contact = new Contact(account, jid, '');
+
+                account.getContactManager().add(contact);
+            }
+        }
+        dialog.close();
+    });
+
+    $('#jsxc-searchresults-button-cancel').on('click',function(e)
+    {
+        dialog.close();
+    });
+}
+
+function getTabeleForSimpleResult(header, items)
+{
+    let html = '<table id="jsxc-searchresults" class="jsxc-searchresults-table"><thead class="jsxc-searchresults-table-head"><tr class="jsxc-searchresults-table-head-row">';
+
+    for (let headitm of header)
+    {
+        html+='<th>'+headitm+'</th>';
+    }
+    html+='</tr></thead><tbody class="jsxc-searchresults-table-body">';
+
+    for (let i=0;i<items.length;i++)
+    {
+        html+='<tr class="jsxc-searchresults-table-body-row '+(i%2===0?'jsxc-searchresults-odd':'jsxc-searchresults-even')+'"><td><input class="jsxc-searchresults-add" type="checkbox" name="jsxc-add-contact" value="'+i+'"/></td>';
+
+        html+='<td>';
+        if ($(items[i]).children('first').length>0)
+        {
+            html+=$(items[i]).children('first').text();
```

Comment:
This allows XSS. We should not build any html from text.

Reply:
changed to use jquery objects and methods now